docs/smart-contracts/meebits_erc721.sol

```diff
@@ -0,0 +1,679 @@
+        address tokenOwner = idToOwner[_tokenId];
+        require(tokenOwner == msg.sender || ownerToOperators[tokenOwner][msg.sender], "Cannot operate.");
+        _;
+    }
+
+    modifier canTransfer(uint256 _tokenId) {
+        address tokenOwner = idToOwner[_tokenId];
+        require(
+            tokenOwner == msg.sender
+            || idToApproval[_tokenId] == msg.sender
+            || ownerToOperators[tokenOwner][msg.sender], "Cannot transfer."
+        );
+        _;
+    }
+
+    modifier validNFToken(uint256 _tokenId) {
+        require(idToOwner[_tokenId] != address(0), "Invalid token.");
+        _;
+    }
+
+    constructor(address _punks, address _glyphs, address payable _beneficiary) {
+        supportedInterfaces[0x01ffc9a7] = true; // ERC165
+        supportedInterfaces[0x80ac58cd] = true; // ERC721
+        supportedInterfaces[0x780e9d63] = true; // ERC721 Enumerable
+        supportedInterfaces[0x5b5e139f] = true; // ERC721 Metadata
+        deployer = msg.sender;
+        punks = _punks;
+        glyphs = _glyphs;
+        beneficiary = _beneficiary;
+    }
+
+    function startSale(uint _price, uint _saleDuration) external onlyDeployer {
+        require(!publicSale);
+        price = _price;
+        saleDuration = _saleDuration;
+        saleStartTime = block.timestamp;
+        publicSale = true;
+        emit SaleBegins();
+    }
+
+    function endCommunityGrant() external onlyDeployer {
+        require(communityGrant);
+        communityGrant = false;
+        emit CommunityGrantEnds();
+    }
+
+    function pauseMarket(bool _paused) external onlyDeployer {
+        require(!contractSealed, "Contract sealed.");
+        marketPaused = _paused;
+    }
+
+    function sealContract() external onlyDeployer {
+        contractSealed = true;
+    }
+
+    //////////////////////////
+    //// ERC 721 and 165  ////
+    //////////////////////////
+
+    function isContract(address _addr) internal view returns (bool addressCheck) {
+        uint256 size;
+        assembly { size := extcodesize(_addr) } // solhint-disable-line
+        addressCheck = size > 0;
+    }
+
+    function supportsInterface(bytes4 _interfaceID) external view override returns (bool) {
+        return supportedInterfaces[_interfaceID];
+    }
+
+    function safeTransferFrom(address _from, address _to, uint256 _tokenId, bytes calldata _data) external override {
+        _safeTransferFrom(_from, _to, _tokenId, _data);
+    }
+
+    function safeTransferFrom(address _from, address _to, uint256 _tokenId) external override {
+        _safeTransferFrom(_from, _to, _tokenId, "");
+    }
+
+    function transferFrom(address _from, address _to, uint256 _tokenId) external override canTransfer(_tokenId) validNFToken(_tokenId) {
+        address tokenOwner = idToOwner[_tokenId];
+        require(tokenOwner == _from, "Wrong from address.");
+        require(_to != address(0), "Cannot send to 0x0.");
+        _transfer(_to, _tokenId);
+    }
```

Comment:
Entry point for ERC721.transferFrom
`require(_to != address(0), "Cannot send to 0x0.");` - Not possible to burn a meebit